README.md
<br/>
<div align="center">
    <h1><a href="https://atheos.io/">Atheos IDE</a>, built on <a href="http://codiad.com/">Codiad</a></h1>
</div>

<div align="center">

[![Codacy Badge](https://api.codacy.com/project/badge/Grade/2c36f9f63b294165b604193efb8cc058)](https://www.codacy.com/gh/Atheos/Atheos?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=Atheos/Atheos&amp;utm_campaign=Badge_Grade)
[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)
[![Quality Gate Status](https://sonarcloud.io/api/project_badges/measure?project=Atheos_Atheos&metric=alert_status)](https://sonarcloud.io/dashboard?id=Atheos_Atheos)

</div>


<div style='margin:0 auto;width:80%;'>

![Screenshot: Atheos](https://www.atheos.io/assets/images/atheos.png "Atheos")

</div>

## About
Atheos is an updated and currently maintained fork of Codiad, a web-based IDE framework with a small footprint and minimal requirements. 

From Codiad:
> "Codiad was built with simplicity in mind, allowing for fast, interactive development without the massive overhead of some of the larger desktop editors. That being said even users of IDE's such as Eclipse, NetBeans and Aptana are finding Codiad's simplicity to be a huge benefit. While simplicity was key, we didn't skimp on features and have a team of dedicated developer actively adding more."

Atheos is expanding on that mentality as much as possible, trying to minimizing it's footprint even further while maximizing functionality and performance. The major goal of Atheos will be to provide users with an easy, fast and fully featured Cloud Based IDE without a large server footprint.

For more information on the project please check out **[the documentation](https://www.atheos.io/docs)** or **[our homepage](http://www.atheos.io)**

## Getting Started
### Manual Installation
Atheos can be installed by placing Atheos in a web accessible folder and pointing your browser to it.
**Ensure that the following folders have write capabilities by your web server:**
```
/config.php
/data
/workspace
/plugins
/themes
```
If any dependencies have not been met, the system will alert you on the installation screen.

For more detailed information see https://www.atheos.io/docs/installation

*Please note: Atheos is primarily developed using with a Debian LAMP Server. If you run into installation issues utilizing another stack, please let us know.*

### Docker
Atheos can  also be installed using the [docker image on DockerHub]([https://hub.docker.com/r/hlsiira/atheos](https://hub.docker.com/r/hlsiira/atheos)).

## Atheos vs Codiad
Atheos is an almost complete rewrite of Codiad, using the latest best practices and technologies available by JavaScript and PHP. All libraries and source code updated, and hundreds of inefficiencies, bugs,  code duplicates, and non standard functions have been addressed. The most notable changes are:
* A complete move from Get requests to Post requests to the server.
* Brand new UX and Theme
* A more complete user permission system 
* Massive reduction in unnecessary/repeated traffic from server to client.
* Complete removal of jQuery and it's plugins
* Built-in Git integration **(WIP)**
* And an incredibly strong emphasis on standardized components and modules throughout the project in order to encourage easier plugin development and user contribution.

Documentation is available at https://www.atheos.io/docs/


content/news.md
---
Description: Atheos is an open source, web-based, cloud IDE and code editor with minimal footprint and requirements, built on top of Codiad by Fluidbyte
Date: 2020-07-10
Cache: false
---
<section>
	<h1>Atheos Cloud Based IDE</h1>

	<p>
		Atheos is a web-based IDE framework with a small footprint and minimal requirements, built on top of Codiad by FluidByte and the Codiad Team.
		Keep up to date with the latest changes and news on <strong><a title="github" href="https://github.com/Atheos/Atheos">Github</a></strong>
	</p>
</section>
<section>
	<h2>Features</h2>
	<p>
		Atheos was built with simplicity in mind, allowing for fast, interactive development without the massive overhead of some of the larger desktop editors. That being said even users of IDE's
		such as Eclipse, NetBeans and Aptana are finding Atheos's simplicity to be a huge benefit. While simplicity was key, we didn't skimp on features and have a team of dedicated developer actively adding more.
	</p>

	<ul class="split_left">
		<li>Support for 40+ languages</li>
		<li>Plugin Library & Marketplace</li>
		<li>Error checking & notifications</li>
		<li>Mutliple user support</li>
		<li>Editor screen splitting</li>
		<li>LocalStorage redundancy</li>
		<li>Advanced searching tools</li>
		<li>Smart auto-complete</li>
	</ul>

	<ul class="split_right">
		<li>Real-Time Collaborative editing</li>
		<li>20+ Syntax color themes</li>
		<li>Completely Open-Source</li>
		<li>Easily customized source</li>
		<li>Runs on your own server</li>
		<li>Quick-Download backups</li>
		<li>Maximum editor screen space</li>
		<li>i18n Language Support</li>
	</ul>

	<div style="clear:both;"></div>
	<p>
		...and with a team of passionate collaborators the list keeps growing.
	</p>
</section>

<section>
	<h2>Interface</h2>

	<p>
		Atheos consists of three panels; a left panel which houses the file manager and active files, center panel which is where editors reside, and a right panel which contains system and other controls. The right &amp; left side panels collapse providing maximum real-estate to the editor.
	</p>

	<img src="assets/images/atheos.jpg" alt="Atheos Screenshot">
</section>

<section>
	<h2>Requirements</h2>

	<p>
		Atheos requires a minimal server installation with Apache2, PHP 7+ and basic R/W access to several directories. Atheos has been tested to work on PHP 5.4, however PHP 7 is where active development takes place. There is no database requirement.
	</p>

	<p>
		The application will work in modern browsers including Chrome, Firefox, and IE9+, though it is developed in Chrome
	</p>
</section>

<section>
	<h2>Getting Started</h2>

	<p>
		Installation of Atheos requires uploading the files and making several directories and a config file writeable. There is no database and no other server components are required. PHP 5.3 or above is recommended, however, the system will work with any version of PHP over v.5.
	</p>

	<p>
		For more information please <strong><a title="installation" href="docs/installation/">read the installation instructions</a></strong> on the Wiki.
	</p>
</section>

<section>
	<h2>Support the Atheos Project</h2>

	<p>
		Hey, this is @HLSiira.
		<br />
		If you want to support Codiad, and all the hard work that they did, please donate to them on <a title="codiad" href="https://www.codiad.com">Codiad</a>.
		I am maintaining my fork of Atheos out of my own personal desire on my own personal time. I am just a dude who likes to program whenever I can as the job that I have has nothing to do with fun.
		If you want to see Atheos be developed further, don't worry, it most likely will as I love working on it and want to see it succeed. The Codiad folks really created a piece of art.
		<br />
		A nice supportive email or message would do me just fine.
	</p>
</section>

home.php
<h3>About Atheos</h3>

<p>
	Atheos is a web-based IDE framework with a small footprint and minimal requirements, built on top of Codiad by FluidByte and the Codiad Team.
</p>

<p>
	Keep up to date with the latest changes and news on <strong><a href="https://github.com/HLSiira/Atheos">Github</a></strong>
</p>

<h3>Features</h3>

<p>
	Atheos was built with simplicity in mind, allowing for fast, interactive development without the massive overhead of some of the larger desktop editors. That being said even users of IDE's
	such as Eclipse, NetBeans and Aptana are finding Atheos's simplicity to be a huge benefit. While simplicity was key, we didn't skimp on features and have a team of dedicated developer actively adding more.
</p>

<ul class="split_left">
	<li>Support for 40+ languages</li>
	<li>Plugin Library & Marketplace</li>
	<li>Error checking & notifications</li>
	<li>Mutliple user support</li>
	<li>Editor screen splitting</li>
	<li>LocalStorage redundancy</li>
	<li>Advanced searching tools</li>
	<li>Smart auto-complete</li>
</ul>

<ul class="split_right">
	<li>Real-Time Collaborative editing</li>
	<li>20+ Syntax color themes</li>
	<li>Completely Open-Source</li>
	<li>Easily customized source</li>
	<li>Runs on your own server</li>
	<li>Quick-Download backups</li>
	<li>Maximum editor screen space</li>
	<li>i18n Language Support</li>
</ul>

<div style="clear:both;"></div>

<br>

<p>
	...and with a team of passionate collaborators the list keeps growing.
</p>

<h3>Interface</h3>

<p>
	Atheos consists of three panels; a left panel which houses the file manager and active files, center panel which is where editors reside, and a right panel which contains system and other controls. The right &amp; left side panels collapse providing maximum real-estate to the editor.
</p>

<img src="assets/images/atheos.png" alt="Atheos Screenshot">

<h3>Requirements</h3>

<p>
	Atheos requires a minimal server installation with Apache2, PHP 5+ and basic R/W access to several directories. There is no database requirement
</p>

<p>
	The application will work in modern browsers including Chrome, Firefox, and IE9+, though it is developed in Chrome
</p>

<h3>Getting Started</h3>

<p>
	Installation of Atheos requires uploading the files and making several directories and a config file writeable. There is no database and no other server components are required. PHP 5.3 or above is recommended, however, the system will work with any version of PHP over v.5.
</p>

<p>
	For more information please <strong><a href="docs/installation">read the installation instructions</a></strong> on the Wiki.
</p>

<h3>Support the Atheos Project</h3>

<p>
	Hey, this is @HLSiira.
</p>
<p>
	If you want to support Codiad, and all the hard work that they did, please donate to them on <a href="https://www.codiad.com">Codiad</a>.
	I am maintaining my fork of Atheos out of my own personal desire on my own personal time. I am just a dude who likes to program whenever I can as the job that I have has nothing to do with fun.
	If you want to see Atheos be developed further, don't worry, it most likely will as I love working on it and want to see it succeed. The Codiad folks really created a piece of art.
</p>
<p>
	But everyone loves a little bit of recognition and I'm not exception.
	A nice supportive email or message would do me just fine.
</p>